### PR TITLE
Remove \OCP\JSON

### PR DESCRIPTION
**Pull request by rullzer**
It should of course be solved properly by moving those functions away from the private class. But I really want to get rid of this in the public interface so nobody that writes app can still use it.

### CODE REVIEW

File: lib/public/JSON.php [hunk 1]
```diff
@@ -1,166 +0,0 @@
...
-	 * updates or deletes anything.
-	 * In cases where you e.g. use an ajax call to load a dialog containing
-	 * a submittable form, you will need to add the requesttoken first as a
-	 * parameter to the ajax call, then assign it to the template and finally
-	 * add a hidden input field also named 'requesttoken' containing the value.
-	 * @deprecated 8.1.0 Use annotation based CSRF checks from the AppFramework instead
-	 *
-	 * @suppress PhanDeprecatedFunction
-	 */
-	public static function callCheck() {
```
Comment (MorrisJobke) [inline lib/public/JSON.php]: Used by following apps:

bildschirmfoto 2018-03-22 um 15 31 58

File: lib/public/JSON.php [hunk 2]
```diff
@@ -1,166 +0,0 @@
...
-	 * Send json success msg
-	 *
-	 * Return a json success message with optional extra data.
-	 * @see \OCP\JSON::error()		for the format to use.
-	 *
-	 * @param array $data The data to use
-	 * @deprecated 8.1.0 Use a AppFramework JSONResponse instead
-	 * @suppress PhanDeprecatedFunction
-	 */
-	public static function success( $data = array() ) {
```
Comment (MorrisJobke) [inline lib/public/JSON.php]: Used by following apps:

bildschirmfoto 2018-03-22 um 15 34 13

File: lib/public/JSON.php [hunk 3]
```diff
@@ -1,166 +0,0 @@
...
-	 *
-	 * Will return the json formatted string:
-	 *
-	 *     {"status":"error","data":{"message":"An error happened", "id":[some value]}}
-	 *
-	 * @param array $data The data to use
-	 * @deprecated 8.1.0 Use a AppFramework JSONResponse instead
-	 * @suppress PhanDeprecatedFunction
-	 */
-	public static function error( $data = array() ) {
```
Comment (MorrisJobke) [inline lib/public/JSON.php]: Used by following apps:

bildschirmfoto 2018-03-22 um 15 37 24

File: lib/public/JSON.php [hunk 4]
```diff
@@ -1,166 +0,0 @@
...
-	 *     {"status":"error","data":{"message":"Application is not enabled."}}
-	 *
-	 * Add this call to the start of all ajax method files that requires
-	 * a specific app to be enabled.
-	 *
-	 * @param string $app The app to check
-	 * @deprecated 8.1.0 Use the AppFramework instead. It will automatically check if the app is enabled.
-	 * @suppress PhanDeprecatedFunction
-	 */
-	public static function checkAppEnabled( $app ) {
```
Comment (MorrisJobke) [inline lib/public/JSON.php]: Used by following apps:

bildschirmfoto 2018-03-22 um 15 37 51

File: lib/public/JSON.php [hunk 5]
```diff
@@ -1,166 +0,0 @@
...
-	 *
-	 *     {"status":"error","data":{"message":"Authentication error."}}
-	 *
-	 * Add this call to the start of all ajax method files that requires
-	 * administrative rights.
-	 *
-	 * @deprecated 8.1.0 Use annotation based ACLs from the AppFramework instead
-	 * @suppress PhanDeprecatedFunction
-	 */
-	public static function checkAdminUser() {
```
Comment (MorrisJobke) [inline lib/public/JSON.php]: Used by following apps:

bildschirmfoto 2018-03-22 um 15 38 22

**Review by MorrisJobke — COMMENTED** [review]
We maybe should ping again the authors of those apps:

* files_opds
* files_reader
* files_snapshots
* piwik
* radio
* rainloop
* user_sql
* W2G2
* working2gether

**Review by MorrisJobke — APPROVED** [review]

File: apps/files_versions/ajax/getVersions.php [hunk 6]
```diff
@@ -47,10 +47,10 @@
 		unset($version['path']);
 	}
 
-	\OCP\JSON::success(array('data' => array('versions' => $versions, 'endReached' => $endReached)));
+	\\OC_JSON::success(array('data' => array('versions' => $versions, 'endReached' => $endReached)));
```
Comment (juliushaertl) [inline apps/files_versions/ajax/getVersions.php]: duplicate backslack :wink:

**Review by juliusknorr — DISMISSED** [review]
There are various places that use `\\OC_JSON` instead of `\OC_JSON`

**Review by MorrisJobke — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Pull request merged**